### PR TITLE
Gift Entry: Batch Mode Enhancements

### LINKED ISSUE #5498: Gift Entry does not enforce "Require Expected Totals Match"
The Process Batch button in Gift Entry allows a batch to be processed regardless of the value of the "Require Expected Totals Match" field in the Batch configuration wizard.

Repro steps:
1. Create a new Batch with 
- Expected Count of Gifts and/or Expected Total Batch Amount populated
- Require Expected Totals Match set to true
2. Enter gifts for the Batch
3. Click "Process Batch" when the count and/or total do not match the expected value(s)

Expected behavior:
- The batch is not able to be processed since the expected count and/or total do not match the running total(s)

Actual behavior:
- The batch is able to be processed

### PR DESCRIPTION
**Pull request by kenlewis-sfdo**
* Gift Entry now includes a progress section that shows running and expected count and total amount.
* The Process Batch button in Gift Entry will now show an error when clicked if the batch requires that totals match and the running totals are not equal to the expected totals. 

# Critical Changes

# Changes
## Gift Entry Updates
- We added progress bars for count and total amount of gifts when entering gifts in a batch. These progress bars will appear when a user sets these values in the Batch Details.

# Issues Closed
Fixes #5498
# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by kenlewis-sfdo** [comment]
**lurch: attach W-037478

**Comment by kenlewis-sfdo** [comment]
**lurch: attach W-038938

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HoXBQA0" target="_blank">W-037478</a>

**Comment by melissabarber** [comment]
@kenlewis-sfdo the labels in this PR look good. Before signing off on this PR, I have questions about two additional labels that should maybe be in this PR.
- Should we also include the warning for if the user has entered batch header values for count or amount, but not checked Require Expected Totals Match?
- I didn't see the help text for the Require Expected Totals Match checkbox in our repo for GE. Could we also include that help text in this work?

**Comment by kenlewis-sfdo** [comment]
**lurch: attach W-038938

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HrByQAK" target="_blank">W-038938</a>

**Comment by kenlewis-sfdo** [comment]
@melissabarber I think we should wait on the warning and treat that as a potential separate ticket - it will need some details about how/when to display that message and what to do next (will the user have to click process batch button twice in that case? or proactive display of warning message?)

Re the help text, do you mean the Inline Help Text for [the field itself](https://github.com/SalesforceFoundation/NPSP/blob/8ea8aca293966d7533e56e404375937fa3576063/src/objects/DataImportBatch__c.object#L405)?  Or help text in the Gift Entry Edit Batch UI?

File: src/labels/CustomLabels.labels [hunk 1]
```diff
@@ -5292,6 +5292,30 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+        <protected>true</protected>
+        <shortDescription>Label for displaying the number of gifts in the batch</shortDescription>
+        <value>Count of Gifts</value>
+    </labels>
+    <labels>
+        <fullName>geBatchGiftsExpectedTotalsMessage</fullName>
+        <categories>Gift Entry</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Notification about number and total of gifts not matching expected.</shortDescription>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels:5308]:
```suggestion
        <shortDescription>Notification about number and total of gifts not matching expected</shortDescription>
```

**Comment by melissabarber** [comment]
> Re the help text, do you mean the Inline Help Text for [the field itself](https://github.com/SalesforceFoundation/NPSP/blob/8ea8aca293966d7533e56e404375937fa3576063/src/objects/DataImportBatch__c.object#L405)? Or help text in the Gift Entry Edit Batch UI?

The help text in the GE batch setup UI.

File: src/labels/CustomLabels.labels [hunk 2]
```diff
@@ -5292,6 +5292,38 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+        <protected>true</protected>
+        <shortDescription>Label for displaying the number of gifts in the batch</shortDescription>
+        <value>Count of Gifts</value>
+    </labels>
+    <labels>
+        <fullName>geBatchGiftsExpectedCountOrTotalMessage</fullName>
+        <categories>Gift Entry</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Message preventing user from processing batch if totals don&apos;t match expected</shortDescription>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels:5308]:
```suggestion
        <shortDescription>Notification about either number or total of gifts not matching expected</shortDescription>
```

File: src/labels/CustomLabels.labels [hunk 3]
```diff
@@ -5292,6 +5292,38 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+        <value>Either the count of gifts or the total gifts doesn&apos;t match the expected values for this batch. These values must match before you can process the batch.</value>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels:5309]:
```suggestion
        <value>The expected doesn't match either the count of gifts or the amount of gifts. These values must match before you can process this batch.</value>
```

**Comment by melissabarber** [comment]
@kenlewis-sfdo @ptewson-sfdo I've reviewed the labels in this PR. I still have an open question to Judi and Ali about if we should have a GitHub Issue for the enforcement of of expected totals. Please also double check with Grace to make sure she's signed off on the UX for this work before merging.

File: src/lwc/utilProgressBar/utilProgressBar.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import {LightningElement, api} from 'lwc';
+
+export default class UtilProgressBar extends LightningElement {
+    @api title;
+
+    @api actualValue;
+    @api expectedValue;
```
Comment (ptewson-sfdo) [inline src/lwc/utilProgressBar/utilProgressBar.js:7]: This is really minor, "expectedValue" seems a little prescriptive for how the progress bar might be used ... "finalValue" or "completedValue" or something? Not sure if is worth focusing on...

**Comment by melissabarber** [comment]
@ptewson-sfdo With the last step of connecting this PR to the GitHub Issue, I'm all good to approve this PR.

File: src/lwc/utilProgressBar/utilProgressBar.js (hunk 4 above)
Comment (kenlewis-sfdo) [reply src/lwc/utilProgressBar/utilProgressBar.js:7]: Yeah, I agree that we could probably come up with something a little more generic for use in other contexts but after "denominator" (which is maybe worse? :) I don't have a lot of great ideas!  Let me know if you have any options and I'm happy to make an update.

**Comment by kenlewis-sfdo** [comment]
fyi @ptewson-sfdo I realized I missed the markup that handles a delete event from the table in the app.  Just pushed that small change and wanted to let you know.

**Pull request merged**